kernel/src/common/regs/README.md

```diff
@@ -0,0 +1,266 @@
+# Register/Bitfield Interface for Tock
+
+This module provides an interface for defining and manipulating memory mapped
+registers and bitfields.
+
+## Defining registers
+
+The module provides three types for working with memory mapped registers: `ReadWrite`,
+`ReadOnly`, and `WriteOnly`, providing read-write, read-only, and write-only functionality,
+respectively.
+
+Defining the registers is similar to the C-style approach, where each register
+is a field in a packed struct:
+
+```rust
+use common::regs::{ReadOnly, ReadWrite, WriteOnly};
+
+#[repr(C, packed)]
+struct Registers {
+    // Control register: read-write
+    // The 'Control' parameter constrains this register to only use fields from
+    // a certain group (defined below in the bitfields section).
+    cr: ReadWrite<u8, Control>,
```

Comment:
It's a little weird to see `u8` here, shouldn't the `Control` type already know that it's size is `u8` from the `register_bitfields!` that defined `Control`? ( https://github.com/helena-project/tock/compare/register-bitfields-bradjc#diff-6efdd170bdb190d142cc1fe9e867dc64R48 )

Reply:
It matches the other registers, where some do not have the `Control` style field.